sqlitecluster/SQLiteNode.cpp

```diff
@@ -1815,7 +1824,7 @@ void SQLiteNode::_changeState(SQLiteNodeState newState) {
 
     if (newState != _state) {
         // First, we notify all plugins about the state change
-        _server.notifyStateChangeToPlugins(_db, newState);
+        _server.notifyStateChangeToPlugins(*pluginDB, newState);
```

Comment:
I think we should move the creation of `pluginDB` into this conditional statement, that way we only create it when we need it.

```cpp
if (newState != _state) {
    // First, we notify all plugins about the state change
        // We create a copy of the database handle here so that the sync node can operate on its handle and the plugin gets
    // its own handle to operate on. This avoids conflicts where the sync thread and the plugin are trying to both run
    // queries at the same time. This also avoids the need to create any share locking between the two.
    pluginDB = new SQLite(_db);
    _server.notifyStateChangeToPlugins(*pluginDB, newState);
```

Reply:
I explained this in slack but will post here for everyone's benefit:

I originally had it there but I moved it to make the lifecycle line up with the life cycle of the SQLiteNode object since I am deleting it in the deconstructor. Additionally it actually doesn’t matter, because functionally it it the same thing. We change the state immediately upon bring a node up. So with the way I have it now, the object probably exists milliseconds longer.

Reply:
Discussed here: https://expensify.slack.com/archives/C094TGUTZ/p1690480820049479